### PR TITLE
Use amo eslint config

### LINKED ISSUE #2667: Use amo eslint config for addons-frontend
See https://github.com/mozilla/eslint-config-amo/

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #2667

This configures addons-frontend to use https://github.com/mozilla/eslint-config-amo/ 

Once this lands all related eslint prs from greenkeeper can be closed.

### CODE REVIEW

File: src/amo/components/AddonMeta.js [hunk 1]
```diff
@@ -54,8 +54,10 @@ export class AddonMetaBase extends React.Component {
           <p className="AddonMeta-text AddonMeta-review-count">
             {reviewCount}
           </p>
-          <Rating className="AddonMeta-Rating" rating={averageRating}
-            readOnly styleName="small" />
+          <Rating className="AddonMeta-Rating"
+            rating={averageRating}
+            readOnly
+            styleName="small" />
```
Comment (tofumatt) [inline src/amo/components/AddonMeta.js]: Just a note that I think these multi-line tags look nicer/align better like:

```jsx
<Rating className="AddonMeta-Rating"
  rating={averageRating}
  readOnly
  styleName="small"
/>
```

it's actually what we (I?) have been doing recently. If eslint is cool with that I'd say change it to that while we're already changing these.

File: src/amo/components/Addon/index.js [hunk 2]
```diff
@@ -220,9 +222,9 @@ export class AddonBase extends React.Component {
     }
 
     return (
-      <ShowMoreCard header={i18n.sprintf(
-        i18n.gettext('About this %(addonType)s'), { addonType }
-      )} className="AddonDescription">
+      <ShowMoreCard
+        header={i18n.sprintf(i18n.gettext('About this %(addonType)s'), { addonType })}
+        className="AddonDescription">
```
Comment (tofumatt) [inline src/amo/components/Addon/index.js]: Same here for the ending `>` alignment.

File: src/amo/components/Categories.js [hunk 3]
```diff
@@ -78,8 +78,9 @@ export class CategoriesBase extends React.Component {
           className="Categories-list"
           ref={(ref) => { this.categories = ref; }}
         >
-          {categories.map((category, index) => (
-            <li className="Categories-list-item" key={`category-${index}`}>
+          {categories.map((category) => (
+            <li className="Categories-list-item"
```
Comment (tofumatt) [inline src/amo/components/Categories.js]: I think if we have newlines in tags at all we should put every attribute on its own line.

File: src/amo/components/Categories.js [hunk 4]
```diff
@@ -78,8 +78,9 @@ export class CategoriesBase extends React.Component {
...
+              key={category.name}>
```
Comment (tofumatt) [inline src/amo/components/Categories.js]: Same here for the ending `>` alignment.

File: src/amo/components/Footer/index.js [hunk 5]
```diff
@@ -45,7 +45,8 @@ export class FooterBase extends React.Component {
               </a>
             </li>
             <li>
-              <a href="#desktop" className="Footer-link Footer-desktop"
+              <a href="#desktop"
```
Comment (tofumatt) [inline src/amo/components/Footer/index.js]: All attrs on newline would be good.

File: src/amo/components/FeaturedAddons/index.js [hunk 6]
```diff
@@ -46,8 +46,11 @@ export class FeaturedAddonsBase extends React.Component {
         <h2 className="FeaturedAddons-header">
           {this.headerForAddonType(addonType)}
         </h2>
-        <SearchResults count={results.length} hasSearchParams={hasSearchParams}
-          loading={loading} results={results} />
+        <SearchResults
+          count={results.length}
+          hasSearchParams={hasSearchParams}
+          loading={loading}
+          results={results} />
```
Comment (tofumatt) [inline src/amo/components/FeaturedAddons/index.js]: Same here for the ending `>` alignment.

File: src/amo/components/LanguagePicker.js [hunk 7]
```diff
@@ -49,10 +49,13 @@ export class LanguagePickerBase extends React.Component {
...
-          ref={(ref) => { this.selector = ref; }} onChange={this.onChange}>
-          {Object.keys(languages).map((locale, index) => (
-            <option key={`${locale}-${index}`} value={locale}>{languages[locale].native}</option>
+        <select className="LanguagePicker-selector"
+          defaultValue={currentLocale}
+          ref={(ref) => { this.selector = ref; }}
+          onChange={this.onChange}>
+          {Object.keys(languages).map((locale) => (
+            <option key={locale}
+              value={locale}>{languages[locale].native}</option>
```
Comment (tofumatt) [inline src/amo/components/LanguagePicker.js]: Same here for the ending `>` alignment.

File: src/amo/components/LanguagePicker.js [hunk 8]
```diff
@@ -49,10 +49,13 @@ export class LanguagePickerBase extends React.Component {
...
           {i18n.gettext('Browse in your language')}
         </h3>
-        <select className="LanguagePicker-selector" defaultValue={currentLocale}
-          ref={(ref) => { this.selector = ref; }} onChange={this.onChange}>
-          {Object.keys(languages).map((locale, index) => (
-            <option key={`${locale}-${index}`} value={locale}>{languages[locale].native}</option>
+        <select className="LanguagePicker-selector"
+          defaultValue={currentLocale}
+          ref={(ref) => { this.selector = ref; }}
+          onChange={this.onChange}>
```
Comment (tofumatt) [inline src/amo/components/LanguagePicker.js]: Same here for the ending `>` alignment.

File: src/amo/components/ScreenShots.js [hunk 9]
```diff
@@ -66,8 +66,11 @@ export default class ScreenShots extends React.Component {
       <div className="ScreenShots">
         <div className="ScreenShots-viewport" ref={(el) => { this.viewport = el; }}>
           <PhotoSwipeGallery
-            className="ScreenShots-list" close={this.onClose} items={formatPreviews(previews)}
-            options={PHOTO_SWIPE_OPTIONS} thumbnailContent={thumbnailContent} />
+            className="ScreenShots-list"
+            close={this.onClose}
+            items={formatPreviews(previews)}
+            options={PHOTO_SWIPE_OPTIONS}
+            thumbnailContent={thumbnailContent} />
```
Comment (tofumatt) [inline src/amo/components/ScreenShots.js]: Same here for the ending `>` alignment.

File: src/amo/components/Search/index.js [hunk 10]
```diff
@@ -70,8 +70,11 @@ export class SearchBase extends React.Component {
     const queryParams = this.props.queryParams ||
       convertFiltersToQueryParams(filters);
     const paginator = count > 0 && hasSearchParams ? (
-      <Paginate LinkComponent={LinkComponent} count={count} currentPage={page}
-        pathname={pathname} queryParams={queryParams} />
+      <Paginate LinkComponent={LinkComponent}
+        currentPage={page}
+        count={count}
+        pathname={pathname}
+        queryParams={queryParams} />
```
Comment (tofumatt) [inline src/amo/components/Search/index.js]: Same here for the ending `>` alignment.

File: src/amo/components/Search/index.js [hunk 11]
```diff
@@ -81,8 +84,11 @@ export class SearchBase extends React.Component {
...
         <SearchContextCard />
         {searchSort}
-        <SearchResults count={count} hasSearchParams={hasSearchParams}
-          filters={filters} loading={loading} pathname={pathname}
+        <SearchResults count={count}
+          filters={filters}
+          hasSearchParams={hasSearchParams}
+          loading={loading}
+          pathname={pathname}
           results={results} />
```
Comment (tofumatt) [inline src/amo/components/Search/index.js]: Same here for the ending `>` alignment.

File: src/amo/components/SearchForm.js [hunk 12]
```diff
@@ -57,19 +57,29 @@ export class SearchFormBase extends React.Component {
     }
 
     return (
-      <form method="GET" action={`/${api.lang}/${api.clientApp}${pathname}`}
-        onSubmit={this.handleSearch} className="SearchForm-form"
+      <form method="GET"
+        action={`/${api.lang}/${api.clientApp}${pathname}`}
+        onSubmit={this.handleSearch}
+        className="SearchForm-form"
         ref={(ref) => { this.form = ref; }}>
```
Comment (tofumatt) [inline src/amo/components/SearchForm.js]: Same here for the ending `>` alignment.

File: src/amo/components/SearchSort/SearchSortLink.js [hunk 13]
```diff
@@ -27,9 +27,8 @@ export default class SearchSortLink extends React.Component {
     const sortURL = `/${pathname}/${queryString({ ...filters, sort })}`;
 
     return (
-      <Link to={sortURL} className={classNames('SearchSortLink', {
-        'SearchSortLink--active': currentSort === sort,
-      })}>
+      <Link to={sortURL}
+        className={classNames('SearchSortLink', { 'SearchSortLink--active': currentSort === sort })}>
```
Comment (tofumatt) [inline src/amo/components/SearchSort/SearchSortLink.js]: Same here for the ending `>` alignment.

File: .eslintrc [hunk 14]
```diff
@@ -1,8 +1,5 @@
 {
-  "env": {
-    "mocha": true,
-  },
-  "extends": "airbnb",
+  "extends": "amo",
```
Comment (tofumatt) [inline .eslintrc]: This is outside the scope of this PR, but we should rename this. I've filed: https://github.com/mozilla/eslint-config-amo/issues/7

File: src/amo/components/AddonMeta.js (hunk 1 above)
Comment (muffinresearch) [inline src/amo/components/AddonMeta.js]: The rule relates to props on the multiline - it doesn't cover closing `>` or `/>` afaik. Will check if there's another rule that can do that.

Comment (muffinresearch) [inline src/amo/components/AddonMeta.js]: Aha looks like there is a rule, I'll add that and use it.

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-closing-bracket-location.md

**Pull request merged**